dcp_prototype/backend/wrangling/migrations/utils/gather_dcp1_data.py

```diff
@@ -0,0 +1,333 @@
+
+    print()
+    group_file_list = gather_group_file_list(file_list)
+
+    print(f"Files in directory to parse: {len(file_list)}")
+
+    for group_files in group_file_list:
+        print(f"Files in group to parse: {len(group_files)}")
+        filequeue = queue.Queue()
+
+        for filename in group_files:
+            filequeue.put(filename)
+
+        threads = []
+        for _ in range(num_threads):
+            thread = threading.Thread(target=consume_file_s3, args=(PREFIX, bucket_name, filequeue, dataset_metadata))
+            thread.start()
+            threads.append(thread)
+
+        filequeue.join()
+        for _ in range(num_threads):
+            filequeue.put(None)
+        for thread in threads:
+            thread.join()
+
+    return dataset_metadata
+
+
+def generate_metadata_structure_from_dir(input_dir, num_threads, dataset_metadata):
+
+    file_list = os.listdir(input_dir)
+    group_file_list = gather_group_file_list(file_list)
+
+    print(f"Files in directory to parse: {len(file_list)}")
+
+    for group_files in group_file_list:
+        print(f"Files in group to parse: {len(group_files)}")
+        filequeue = queue.Queue()
+
+        for filename in group_files:
+            filequeue.put(os.path.join(input_dir, filename))
+
+        threads = []
+        for _ in range(num_threads):
+            thread = threading.Thread(target=consume_file_local, args=(filequeue, dataset_metadata))
+            thread.start()
+            threads.append(thread)
+
+        filequeue.join()
+        for _ in range(num_threads):
+            filequeue.put(None)
+        for thread in threads:
+            thread.join()
+
+    return dataset_metadata
+
+
+def generate_metadata_structure_from_targz(input_source, num_threads, dataset_metadata):
+
+    index = 0
+    print("open ", input_source)
+    with tarfile.open(input_source, "r:gz") as tar:
+        print("get members")
+        members = tar.getmembers()
+        print("get members done")
+        entity_group = []
+        link_group = []
+
+        for index, member in enumerate(members):
+            buf = tar.extractfile(member).read()
+            if member.name.endswith(".json"):
+                entity_group.append((member.name, buf))
+            else:
+                link_group.append((member.name, buf))
+
+            if index > 0 and index % 1000 == 0:
+                print(index, end=" ", flush=True)
+
+    print(index)
+    group_file_list = [entity_group, link_group]
+    print(f"Files in directory to parse: {len(members)}")
+
+    for group_files in group_file_list:
+        print(f"Files in group to parse: {len(group_files)}")
+        index = 0
+        for filename, buf in group_files:
+            object_data = json.loads(buf)
+            dataset_metadata.add_entity(filename, object_data)
+            index += 1
+            if index % 1000 == 0:
+                print("processed ", index)
+
+    return dataset_metadata
+
+
+def main():
```

Comment:
This section of code seems to be exactly the same as `migrate_dcp_one_to_dcp_two.py`?

Reply:
Oh yeah,.  I had meant to remove all that.   I'll do that now.